src/constants.ts

```diff
@@ -142,3 +142,14 @@ export const G_MAX_FORM_KEYS = Object.freeze([
   SpeciesFormKey.GIGANTAMAX_RAPID,
   SpeciesFormKey.GIGANTAMAX_SINGLE,
 ]) as readonly string[];
+
+/** Each wave, all unfainted Pokemon gain this much happiness. Used in {@link NextEncounterPhase} */
+export const FRIENDSHIP_GAIN_PER_WAVE = 1;
+/** Value for how much friendship a Pokemon gains on leveling up. Used in {@link LevelUpPhase} */
+export const FRIENDSHIP_GAIN_PER_LEVEL_UP = 10;
+/** After this point, friendship gain is halved (rounded down). Used in {@linkcode addFriendship} */
+export const FRIENDSHIP_GAIN_CUTOFF = 150;
+/** Additional friendship gained from rare candy. Used in {@link PokemonLevelIncrementModifier} */
+export const FRIENDSHIP_FROM_CANDY = 5;
```

Comment:
```suggestion
export const FRIENDSHIP_GAIN_FROM_RARE_CANDY = 5;
```
Rename through VSCode.